specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-04-01/machineLearningServices.json

```diff
@@ -4701,6 +4717,260 @@
+        }
+      }
+    },
+    "CustomService": {
+      "type": "object",
+      "description": "Specifies the custom service configuration",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the Custom Service"
+        },
+        "image": {
+          "$ref": "#/definitions/Image",
+          "description": "Describes the Image Specifications"
+        },
+        "environmentVariables": {
+          "type": "object",
+          "description": "Environment Variable for the container",
+          "additionalProperties": {
+            "$ref": "#/definitions/EnvironmentVariable"
+          }
+        },
+        "docker": {
+          "description": "Describes the docker settings for the image",
+          "$ref": "#/definitions/Docker",
+          "x-nullable": true
+        },
+        "endpoints": {
+          "type": "array",
+          "description": "Configuring the endpoints for the container",
+          "items": {
+            "$ref": "#/definitions/Endpoint"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "volumes": {
+          "type": "array",
+          "description": "Configuring the volumes for the container",
+          "items": {
+            "$ref": "#/definitions/VolumeDefinition"
+          },
+          "x-ms-identifiers": [
+            "source",
+            "target"
+          ]
+        }
+      },
+      "additionalProperties": true
+    },
+    "Image": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "default": "docker",
+          "enum": [
+            "docker",
+            "azureml"
+          ],
+          "type": "string",
+          "title": "Type of the image",
+          "description": "Type of the image. Possible values are: docker - For docker images. azureml - For AzureML images",
+          "x-ms-enum": {
+            "name": "ImageType",
+            "modelAsString": true
+          }
+        },
+        "reference": {
+          "type": "string",
+          "description": "Image reference"
+        }
+      },
+      "additionalProperties": true
+    },
+    "EnvironmentVariable": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "local"
+          ],
+          "default": "local",
+          "title": "Type of Environment Variable",
+          "description": "Type of the Environment Variable. Possible values are: local - For local variable",
+          "type": "string",
+          "x-ms-enum": {
+            "name": "EnvironmentVariableType",
+            "modelAsString": true
+          }
+        },
+        "value": {
+          "type": "string",
+          "description": "Value of the Environment variable"
+        }
+      },
+      "additionalProperties": true
+    },
+    "Docker": {
+      "type": "object",
+      "properties": {
+        "privileged": {
```

Comment:
>privileged [](http://example.com/codeflow?start=9&length=10)

It is always recommended to use enum instead of boolean, like here:

"mode" as enum with values: "privileged", "unprivileged".

That makes it extensible in future without causing ay breaking changes. Unless you are absolutely sure, it will always be just these values, needing it as boolean.